site/blog/2022-11-04.mdx

```diff
@@ -0,0 +1,123 @@
+
+In this post, we're going to discover an IaSQL module that's built to make deploying to ECS, simplified. Most of the details for deploying a container to ECS are the same (load balancers, security groups, IAM roles, etc), and we have created the `ecs_simplified` module for you so that you can give it your codebase and get your app deployed to ECS **in just 11 minutes**. All of those needed resources are going to be created automatically in your AWS account, and you'll have full access over them either through the AWS dashboard, or through your IaSQL database.
+
+## ** High-level vs Low-level IaSQL Modules **
+In IaSQL, you are able to do cloud actions both on a high-level and a low-level abstraction layer. It means that you can go with both of these two approaches:
+1. Deploy the folder `examples/ecs-fargate/prisma/app` of the repo `https://github.com/iasql/iasql-engine/` to ECS on my AWS account
+2. Create a security group named `allow-all` attached to the VPC with ID `vpc-f56cff91`. Then add a rule to it allowing all incoming traffic on port `80` from `0.0.0.0/0`.
+
+The first statement, is described in a high-level abstraction layer. The higher the level, the less detail. So it doesn't say anything about the underlying steps that should be taken in order for the statement to be executed. But the second statement is giving all the details necessary to create a new security group.
+
+The flexibility of IaSQL– which makes it really awesome– is that you can create both the high-level and the low-level modules for it. Both of the above statements are already possible using IaSQL modules. The first one using a module named `aws_ecs_simplified` and the latter using the `aws_security_group` module. Let's take a peek on the actual implementations really quick before we continue our main topic.
+
+```sql title="Deploy the folder 'examples/ecs-fargate/prisma/app' of the 'iasql-engine' repo to ECS on my AWS account"
+-- Install the "high-level" aws_ecs_simplified IaSQL module
+SELECT iasql_install('aws_ecs_simplified');
+-- Create the load balancer, ECS service, task definition, etc. All ready afterwards for us to build and push to deploy the app.
+INSERT INTO ecs_simplified (app_name, app_port, image_tag, public_ip) VALUES ('simple-express', 8088, 'latest', true);
+SELECT iasql_apply();
+-- Use the "high-level" ecr_build function to build the folder "examples/ecs-fargate/prisma/app" from "iasql-engine" Github repo and push it to ECR
+SELECT ecr_build(
+  'https://github.com/iasql/iasql-engine/',
+  (SELECT id FROM repository WHERE repository_name = 'simple-express-repository')::varchar(255),
+  './examples/ecs-fargate/prisma/app',
+  'ghp_XXX', -- replace your github personal access token here
```

Comment:
Public repos shouldn't need the github access token, right? Can this be optional?

Reply:
Oh that's a really messy AWS CodeBuild stuff. I tried a bunch of times without any personal access tokens and it cannot even pull the public repos. God knows why it needs a PAT for just cloning a repo. I'll try again to see if it can be done without a PAT (in case it was because of IAM role or something else and I've misjudged).

Reply:
In the worst case, we can sign up for a Github account and create a PAT (with only access to the public repos) and put it here? Or maybe put it on the engine?

Reply:
Wait, I have another idea: we can set the source provider in the CodeBuild to "No Source" and do a normal `git clone` at the first of the `buildspec.yaml` file. In case a PAT is provided by the user, we can go with the normal flow.

Reply:
I tested that idea and it worked. I'll change the `ecr_build` to use that behavior.